src/Fixer/Casing/LowercaseConstantsFixer.php

```diff
@@ -24,8 +27,33 @@
  *
  * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
  */
-final class LowercaseConstantsFixer extends AbstractFixer
+final class LowercaseConstantsFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface
 {
+    /**
+     * Hold the function that will be used to convert the constants.
+     *
+     * @var callable
+     */
+    private $fixFunction;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->fixFunction = static function ($string) {
+            return strtolower($string);
+        };
+
+        if ('upper' === $this->configuration['case']) {
```

Comment:
I think `if else` would be a better fit, so the first lambda is not created if we run with the `upper` case

Reply:
Ok, I thought it was a good idea as 'lower' will be the default value.
Going to do the change now.

Reply:
Fixed, waiting for the tests.